### ISSUE TITLE
modifier breaks with parenthesis and does not work with interpolation (v8)

### ISSUE THREAD

**Issue by jd1378** [nuxt-modules/i18n#2225]
### Environment

------------------------------
- Operating System: Linux
- Node Version:     v16.17.0
- Nuxt Version:     3.6.2
- Nitro Version:    2.5.2
- Package Manager:  yarn@1.22.19
- Builder:          vite
- User Config:      telemetry, modules, i18n
- Runtime Modules:  @nuxtjs/i18n@8.0.0-beta.13
- Build Modules:    -
------------------------------

### Reproduction

https://codesandbox.io/p/github/nuxt/starter/csb-v7tdxl/draft/dry-star?file=/i18n.config.ts:8,41

### Describe the bug

the test is straightforward
I use `{{ t("testNum", { param: "something" }) }}` to run the following translations and I have documented what happens, you can see the reproduction as well:
```ts
// i18n.config.ts
export default defineI18nConfig(() => ({
  legacy: false,
  locale: "en",
  messages: {
    en: {
      foo: "foo",
      fooParam: "{param}",
      test1: "Welcome {param}", // works
      test2: "Welcome @.upper:foo", // works and shows whats inside foo
      test3: "Welcome @.upper:fooParam", // works interestingly
      test4: "Welcome (@.upper:foo)", // broken
      test5: "Welcome (@.upper:foo )", // works brokenly
      test6: "Welcome (@.upper:{param})", // broken
      test7: "Welcome (@.upper:{param} )", // broken
    },
  },
}));
```
output:
```
test1: Welcome something
test2: Welcome FOO
test3: Welcome SOMETHING
test4: Welcome (
test5: Welcome (FOO )
test6: Welcome ()
test7: Welcome ( )
```

**Comment by kazupon** [comment]
Thank you for your reporting!

This is bug of vue-i18n message compiler.
I'll try to fix.

**Comment by jd1378** [comment]
@kazupon any eta on this ? can I help with this issue somehow ? this is an important issue to me and I'm willing to contribute, But I need to know where to look for and where to make pr against
thanks a lot

**Comment by kazupon** [comment]
I've fixed in intlify/vue-i18n-next#1506
You can try to use nuxt i18n edge channel version.
so close.
